chainercv/datasets/cityscapes/cityscapes_semantic_segmentation_dataset.py

```diff
@@ -0,0 +1,108 @@
+            will be replaced with :obj:`-1` in the :meth:`get_example` method.
+            The default value is :obj:`True`.
+
+    """
+
+    def __init__(self, data_dir=None, label_mode=None, split='train',
+                 ignore_labels=True):
+        if data_dir is None:
+            data_dir = download.get_dataset_directory(
+                'pfnet/chainercv/cityscapes')
+        if label_mode not in ['fine', 'coarse']:
+            raise ValueError('\'label_mode\' argment should be eighter '
+                             '\'fine\' or \'coarse\'.')
+
+        img_dir = os.path.join(data_dir, os.path.join('leftImg8bit', split))
+        resol = 'gtFine' if label_mode == 'fine' else 'gtCoarse'
+        label_dir = os.path.join(data_dir, resol)
+        if not os.path.exists(img_dir) or not os.path.exists(label_dir):
+            raise ValueError(
+                'Cityscapes dataset does not exist at the expected location.'
+                'Please download it from https://www.cityscapes-dataset.com/.'
+                'Then place directory leftImg8bit at {} and {} at {}.'.format(
+                    os.path.join(data_dir, 'leftImg8bit'), resol, label_dir))
+
+        self.ignore_labels = ignore_labels
+
+        self.label_paths = list()
+        self.img_paths = list()
+        city_dnames = list()
+        for dname in glob.glob(os.path.join(label_dir, '*')):
+            if split in dname:
+                for city_dname in glob.glob(os.path.join(dname, '*')):
+                    for label_path in glob.glob(
+                            os.path.join(city_dname, '*_labelIds.png')):
+                        self.label_paths.append(label_path)
+                        city_dnames.append(os.path.basename(city_dname))
+        for city_dname, label_path in zip(city_dnames, self.label_paths):
+            label_path = os.path.basename(label_path)
+            img_path = label_path.replace(
+                '{}_labelIds'.format(resol), 'leftImg8bit')
+            img_path = os.path.join(img_dir, city_dname, img_path)
+            self.img_paths.append(img_path)
+
+    def __len__(self):
+        return len(self.img_paths)
+
+    def get_example(self, i):
+        """Returns the i-th example.
+
+        Returns a color image and a label image. The color image is in CHW
+        format and the label image is in HW format.
+
+        Args:
+            i (int): The index of the example.
+
+        Returns:
+            tuple of a color image and a label whose shapes are (3, H, W) and
+            (H, W) respectively. H and W are height and width of the image.
+            The dtype of the color image is :obj:`numpy.float32` and
+            the dtype of the label image is :obj:`numpy.int32`.
+
+        """
+        img = read_image(self.img_paths[i])
+        label_orig = read_image(
+            self.label_paths[i], dtype=np.int32, color=False)[0]
+        H, W = label_orig.shape
```

Comment:
I think we can delete this and use `np.ones(label_orgi.shape, dtype=np.int32) *  -1` instead for line 102.